popde/density_estimate.py

```diff
@@ -108,18 +108,83 @@ def prepare_data(self):
             self.transf_data = self.data
 
         if self.stdize:
-            std_transf = ['stdize' for dim in self.ndim]
+            std_transf = ['stdize'] *self.ndim
             self.stds = np.std(self.transf_data, axis=0)  # record the stds
             self.std_data = transf.transform_data(self.transf_data, std_transf)
         else:
-            self.stds = None
             self.std_data = self.transf_data
 
         if self.rescale is not None:
             self.kde_data = transf.transform_data(self.std_data, self.rescale)
         else:
             self.kde_data = self.std_data
 
+
+    def evaluate_with_transf(self, points):
+        """
+        Transforms the input points in the same way as the KDE training data,
+        evaluates the KDE on the transformed points and adjusts the KDE
+        values by the Jacobian of the transformations.
+
+        Parameters:
+        -----------
+        points : array-like, shape (n_samples, n_features)
+            The original parameter space points to be transformed and evaluated.
+
+        Returns:
+        --------
+        kde_vals : array-like, shape (n_samples,)
+        The KDE values adjusted by the Jacobian of the transformations.
+        """
+        if self.input_transf is not None:
+            transf_points = transf.transform_data(points, self.input_transf)
+        else:
+            transf_points = points
+
+        # Divide each parameter by the std of the training data
+        if self.stdize:
+            std_points = transf.transform_data(transf_points, 1.0/self.stds)
+        else:
+            std_points = transf_points
+
+        # Rescale each parameter
+        if self.rescale is not None:
+            transf_data = transf.transform_data(std_points, self.rescale)       
+        else:
+            transf_data = std_points
+
+        #Evaluate kde on transform points
+        kde_vals = self.evaluate(transf_data)
+
+        # Jacobian of transform
+        for i, option in enumerate(self.input_transf):
+            if option in['log', 'ln']:
+                input_Jacobian = 1.0 / points[:, i]
+            elif option == 'exp':
+                input_Jacobian = np.exp(points[:, i])
+            elif option in ('none', 'None'):
+                input_Jacobian  = 1.0 
+            else:
+                raise ValueError(f"Invalid transformation option at index {i}: {option}")
+
+            #standardize using stds of training data 
+            if self.stdize:
+                Jacobian_std_factor = self.stds[i]
+            else:
+                Jacobian_std_factor = 1.0
+
+            #rescale based on self.rescale list in each dimension
+            if self.rescale is not None:
+                Jacobian_rescale_factor = self.rescale[i]
+            else:
+                Jacobian_rescale_factor = 1.0
+
+            Total_Jacobian =  input_Jacobian * Jacobian_std_factor * Jacobian_rescale_factor
+            kde_vals   *= Total_Jacobian
```

Comment:
```suggestion
                rescale_Jacobian = self.rescale[i]
            else:
                rescale_Jacobian = 1.0

            kde_vals *= input_Jacobian * std_Jacobian * rescale_Jacobian
```
I don't think it is necessary to define total_Jacobian here in fact